staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -20,6 +20,9 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/go-openapi/spec"
+	"github.com/go-openapi/strfmt"
+	"github.com/go-openapi/validate"
 	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
```

Comment:
I would move the validation logic into `k8s.io/apiextensions-apiserver/pkg/apiserver/validation`, to keep the actual API types separate from the "business logic".